cli/internal/k8s/common.go

```diff
@@ -153,21 +152,13 @@ func init() {
 func getRestConfig() *rest.Config {
 	message.Debug("k8s.getRestConfig()")
 
-	// use the KUBECONFIG context if it exists
-	configPath := os.Getenv("KUBECONFIG")
-	if configPath == "" {
-		// use the current context in the default kubeconfig in the home path of the user
-		homePath, err := os.UserHomeDir()
-		if err != nil {
-			message.Fatal(nil, "Unable to load the current user's home directory")
-		}
-		configPath = homePath + "/.kube/config"
-	}
-
-	config, err := clientcmd.BuildConfigFromFlags("", configPath)
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
This is a pretty critical change (I think for the better) to the core of how we talk to the control plane that is well outside the scope of a PR for simplifying tests.  Would like to see more code comments here as to what each step is doing and a comment in the PR on why this change was needed here vs it's own PR as changing core code behavior for a tests is usually and anti-pattern.

Reply:
I can pull it back out into its own PR. It was originally its own PR, but it got incorporated here later.

Reply:
I think it can be fine here is you can explain in this PR what's going on.  Also some code comments per line would be nice here because it's k8s magic block box right now.